tests/cupy_tests/linalg_tests/test_eigenvalue.py

```diff
@@ -27,14 +27,32 @@ def _get_hermitian(xp, a, UPLO):
     reason='eigensolver not added until ROCm 4.2.0')
 class TestEigenvalue:
 
-    @testing.for_all_dtypes(no_float16=True, no_complex=True)
+    @testing.for_all_dtypes()
     @testing.numpy_cupy_allclose(rtol=1e-3, atol=1e-4, contiguous_check=False)
     def test_eigh(self, xp, dtype):
-        a = xp.array([[1, 0, 3], [0, 5, 0], [7, 0, 9]], dtype)
+        if xp == numpy and dtype == numpy.float16:
+            # NumPy's eigh does not support float16
+            _dtype = 'f'
+        else:
+            _dtype = dtype
+        a = xp.array([[1, 0, 3], [0, 5, 0], [7, 0, 9]], _dtype)
         w, v = xp.linalg.eigh(a, UPLO=self.UPLO)
-        # NumPy, cuSOLVER, rocSOLVER all sort in ascending order,
-        # so they should be directly comparable
-        return w, v
+
+        # Changed the verification method to check if Av and vw match, since
+        # the eigenvectors of eigh() with CUDA 11.6 are mathematically correct
+        # but may not match NumPy.
+        if self.UPLO == 'U':
+            A = xp.triu(a) + xp.transpose(xp.triu(a, 1))
+        else:
+            A = xp.tril(a) + xp.transpose(xp.tril(a, -1))
```

Comment:
This looks correct and more readable than current `_get_hermitian`.  How about fixing `_get_hermitian` instead?
```python
def _get_hermitian(xp, a, UPLO):
    if UPLO == 'U':
        return xp.triu(a) + xp.triu(a, 1).swapaxes(-2, -1).conj()
    else:
        return xp.tril(a) + xp.tril(a, -1).swapaxes(-2, -1).conj()
```

Reply:
That's a good idea. I'll change `_get_hermitian` to the above and use it in `test_eigh` as well.